sys/riscv/riscv/machdep.c

```diff
@@ -1119,6 +1120,13 @@ initriscv(struct riscv_bootparams *rvbp)
 
 	/* Bootstrap enough of pmap to enter the kernel proper */
 	kernlen = (lastaddr - KERNBASE);
+	/* Check if bootloader clobbered part of the kernel with the DTB. */
+	KASSERT(rvbp->dtbp_phys + rvbp->dtbp_size < rvbp->kern_phys ||
+		rvbp->dtbp_phys > rvbp->kern_phys + kernlen,
+	    ("FDT (%lx-%lx) and kernel (%lx-%lx) overlap", rvbp->dtbp_phys,
+		rvbp->dtbp_phys + rvbp->dtbp_size, rvbp->kern_phys,
+		rvbp->kern_phys + kernlen));
+
```

Comment:
This belongs inside `fake_preload_metadata`, as loader(8) does not use these fields (see https://reviews.freebsd.org/D24912).